homeassistant/components/flume/sensor.py

```diff
@@ -0,0 +1,246 @@
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit the value is expressed in."""
+        return self._unit_of_measurement
+
+    def update(self):
+        """Get the latest data and updates the states."""
+        self.Flume.update()
+        self._state = self.Flume.value
+
+
+class FlumeAuth:
+    """Get the Authentication Bearer, User ID and list of devices from Flume API."""
+
+    def __init__(self, username, password, client_id, client_secret):
+        """Initialize the data object."""
+        self._username = username
+        self._password = password
+        self._client_id = client_id
+        self._client_secret = client_secret
+        self._token = self.getToken()
+        self._user_id = self.getUserId()
+        self._bearer = self.getBearer()
+        self._devices = self.getDevices()
+
+    def getToken(self):
+        """Return authorization token for session."""
+        url = "https://api.flumetech.com/oauth/token"
+        payload = (
+            '{"grant_type":"password","client_id":"'
+            + self._client_id
+            + '","client_secret":"'
+            + self._client_secret
+            + '","username":"'
+            + self._username
+            + '","password":"'
+            + self._password
+            + '"}'
+        )
+        headers = {"content-type": "application/json"}
+
+        response = requests.request("POST", url, data=payload, headers=headers)
+
+        _LOGGER.debug("Token Payload: %s", payload)
+        _LOGGER.debug("Token Response: %s", response.text)
+
+        if response.status_code == 200:
+            return json.loads(response.text)["data"]
+        else:
+            raise Exception(
+                "getToken Response Code not Successful. Returned {}".format(
+                    response.status_code
+                )
+            )
+
+    def getUserId(self):
+        """Return User ID for authorized user."""
+        json_token_data = self._token[0]
+        return json.loads(
+            base64.b64decode(json_token_data["access_token"].split(".")[1])
+        )["user_id"]
+
+    def getBearer(self):
+        """Return Bearer for Authorized session."""
+        return self._token[0]["access_token"]
+
+    def getDevices(self):
+        """Return all available devices from Flume API."""
+        url = "https://api.flumetech.com/users/" + str(self._user_id) + "/devices"
+        querystring = {"user": "false", "location": "false"}
+        headers = {"authorization": "Bearer " + self._bearer + ""}
+        response = requests.request("GET", url, headers=headers, params=querystring)
+
+        _LOGGER.debug("getDevices Response: %s", response.text)
+
+        if response.status_code == 200:
+            return json.loads(response.text)["data"]
+        else:
+            raise Exception(
+                "getDevices Response Code not Successful. Returned {}".format(
```

Comment:
Exception message should be updated for production.

Something like `Impossible to retreive devices. Response code returned : {}`